app/lib/app/SecvisogramPage/View/FormEditor/editors/GenericEditor/Attributes/CVSS3Attribute.js

```diff
@@ -52,26 +53,23 @@ export default function CVSSV3Attribute({
   )
 
   /** @type {(childName: string, options: string[], disableClearable: boolean) => any} */
-  function dropdownFor(childName, options, disableClearable = true) {
-    return (
-      <DropdownAttribute
-        label={childName.charAt(0).toUpperCase() + childName.substring(1)}
-        description=""
-        options={options}
-        isEnum={true}
-        instancePath={instancePath.concat([childName])}
-        value={(value || {})[childName] || ''}
-        property={property}
-        disabled={disabled}
-        disableClearable={disableClearable}
-      />
+  function dropdownFor(childName, options, disableClearable = false) {
+    const childValue = /** @type {string} */ ((value || {})[childName]) || ''
+    return cvssDropdown(
+      instancePath,
+      childName,
+      childValue,
+      options,
+      property,
+      disabled,
+      disableClearable
```

Comment:
I might be wrong, but don't we use the same function also in CVSS 2? If so, I guess that should rather be part of a common files, as I don't want to copy the function as soon as CVSS 4.0 comes along the way...

Reply:
You're not wrong. `cvssDropdown` is already a shared function. If we factor in other values than `childName`, `options` and `disableClearable` we could get rid of the repeated `dropdownFor` but would have to repeatedly pass the additional arguments whenever calling the function.

So the seemingly same function wraps the differing arguments for one CVSS object.

Reply:
I'm not sure whether I was able to follow you: My question was: "Is there a difference between the `dropdownFor` function in CVSS2Attributes (L62-74) and CVSS3Attributes (L56-68)?" If so, what is different (as I wasn't able to spot it) and if not, is there a specific reason, why this function is not extracted and imported from a single file into both?

Reply:
No, there is no difference.

The `dropdownFor` function is a helper to wrap arguments from different scopes. If we outsource it completely (like `cvssDropdown` already is), we will need to also pass the shared (within one CVSSAttribute object) parameters whenever calling it. Replacing all the calls like
`dropdownFor('version', ['3.0', '3.1'], true)`
with sth. like
`cvssDropdown(instancePath, 'version', getChildValue('version'),  ['3.0', '3.1'], property, disabled, true)`
where `instancePath`, `property` and `disabled` will always be the same and `getChildValue` is an additional function (that will look the same again).
I totally understand your preference to avoid repetition but on the other hand, the second option becomes quite bloated.

If you prefer it that way, feel free to merge in https://github.com/secvisogram/secvisogram/pull/417

Reply:
@valentindey Thank you for that explanation - that was the piece I was missing.

I'll reject #417 as you are right that the current implementation seems to be the best variant.